azurerm/internal/services/sentinel/sentinel_automation_rule_resource_test.go

```diff
@@ -0,0 +1,359 @@
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (r SentinelAutomationRuleResource) Exists(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	client := clients.Sentinel.AutomationRulesClient
+
+	id, err := parse.AutomationRuleID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp, err := client.Get(ctx, id.ResourceGroup, sentinel.OperationalInsightsResourceProvider, id.WorkspaceName, id.Name); err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	return utils.Bool(true), nil
+}
+
+func (r SentinelAutomationRuleResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_sentinel_automation_rule" "test" {
+  name                       = "%s"
+  log_analytics_workspace_id = azurerm_log_analytics_solution.sentinel.workspace_resource_id
+  display_name               = "acctest-SentinelAutoRule-%d"
+  order                      = 1
+
+  action_incident {
+    order  = 1
+    status = "Active"
+  }
+}
+`, template, r.uuid, data.RandomInteger)
+}
+
+func (r SentinelAutomationRuleResource) complete(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+data "azurerm_client_config" "current" {}
+
+data "azuread_service_principal" "securityinsights" {
+  display_name = "Azure Security Insights"
+}
+
+resource "azurerm_role_assignment" "sentinel" {
+  scope                = azurerm_resource_group.test.id
+  role_definition_name = "Azure Sentinel Automation Contributor"
+  principal_id         = data.azuread_service_principal.securityinsights.object_id
+}
+
+resource "azurerm_template_deployment" "testconnection" {
+  name                = "testconnection"
+  resource_group_name = azurerm_resource_group.test.name
+  deployment_mode     = "Incremental"
+  template_body       = <<TEMPLATE
+	{
+		"$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+		"contentVersion": "1.0.0.0",
+		"parameters": {
+			"connections_azuresentinel_name": {
+				"defaultValue": "azuresentinel",
+				"type": "String"
+			}
+		},
+		"variables": {},
+		"resources": [
+			{
+				"type": "Microsoft.Web/connections",
+				"apiVersion": "2016-06-01",
+				"name": "[parameters('connections_azuresentinel_name')]",
+				"location": "${azurerm_resource_group.test.location}",
+				"kind": "V1",
+				"properties": {
+					"displayName": "test",
+					"customParameterValues": {},
+					"api": {
+						"id": "[concat('/subscriptions/${data.azurerm_client_config.current.subscription_id}/providers/Microsoft.Web/locations/${azurerm_resource_group.test.location}/managedApis/', parameters('connections_azuresentinel_name'))]"
+					}
+				}
+			}
+		]
+	}
+  TEMPLATE
+}
+
+resource "azurerm_template_deployment" "testlogicapp" {
```

Comment:
and here

Reply:
This piece of template is to provision a `azurerm_logic_app_workflow` with `azurerm_logic_app_trigger_custom` embedded. It looks like we can do the same by using these existing resources as below:

```hcl
resource "azurerm_logic_app_workflow" "test" {
  name                = "testworkflow1"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
}

resource "azurerm_logic_app_trigger_custom" "test" {
  name         = "When_Azure_Sentinel_incident_creation_rule_was_triggered"
  logic_app_id = azurerm_logic_app_workflow.test.id

  body = <<BODY
{og
  "type": "ApiConnectionWebhook",
  "inputs": {
    "body": {
      "callback_url": "@{listCallbackUrl()}"
    },
    "host": {
      "connection": {
        "name":  "/subscriptions/${data.azurerm_client_config.current.subscription_id}/resourceGroups/${azurerm_resource_group.test.name}/providers/Microsoft.Web/connections/azuresentinel"
      }
    },
    "path": "/incident-creation"
  }
}
BODY

  depends_on = [azurerm_template_deployment.testconnection]
}
```

However, this ends up with a logic app workflow non-functional (which is greyed out in the logic app designer interface, and has a red mark complaining *Connector not found*). By comparing the template (logic app code), the difference might be the format, where the logic app might ask for the certain parameter to be declared and defined. Then it means we will need customized parameter definition for the `azurerm_logc_app_workflow` resource, which should be resolved by #12314.

Reply:
**The non-functional one**

![image](https://user-images.githubusercontent.com/7970698/123928658-b0515300-d9c0-11eb-926d-cdd0ff245031.png)

*code view*

```
{
    "definition": {
        "$schema": "https://pluginsdk.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#",
        "actions": {},
        "contentVersion": "1.0.0.0",
        "triggers": {
            "When_Azure_Sentinel_incident_creation_rule_was_triggered": {
                "inputs": {
                    "body": {
                        "callback_url": "@{listCallbackUrl()}"
                    },
                    "host": {
                        "connection": {
                            "name": "/subscriptions/****/resourceGroups/acctestRG-sentinel-999/providers/Microsoft.Web/connections/azuresentinel"
                        }
                    },
                    "path": "/incident-creation"
                },
                "type": "ApiConnectionWebhook"
            }
        }
    },
    "parameters": {}
}
```

**THe functional one**

![image](https://user-images.githubusercontent.com/7970698/123928719-bd6e4200-d9c0-11eb-8602-68921368f930.png)

*code view*

```
{
    "definition": {
        "$schema": "https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#",
        "contentVersion": "1.0.0.0",
        "outputs": {},
        "parameters": {
            "$connections": {
                "defaultValue": {},
                "type": "Object"
            }
        },
        "triggers": {
            "When_Azure_Sentinel_incident_creation_rule_was_triggered": {
                "inputs": {
                    "body": {
                        "callback_url": "@{listCallbackUrl()}"
                    },
                    "host": {
                        "connection": {
                            "name": "@parameters('$connections')['azuresentinel']['connectionId']"
                        }
                    },
                    "path": "/incident-creation"
                },
                "type": "ApiConnectionWebhook"
            }
        }
    },
    "parameters": {
        "$connections": {
            "value": {
                "azuresentinel": {
                    "connectionId": "/subscriptions/****/resourceGroups/acctestRG-sentinel-888/providers/Microsoft.Web/connections/azuresentinel",
                    "connectionName": "azuresentinel",
                    "id": "/subscriptions/****/providers/Microsoft.Web/locations/westeurope/managedApis/azuresentinel"
                }
            }
        }
    }
}
```

Reply:
Correct above findings, the reason why the former doesn't work is because of the setting inside the `triggers.When***.inputs.host.connection`, it should be set into `@parameters('$connections')['azuresentinel']['connectionId']`, rather than the evaluated value, I have no idea why thoug :joy: